File: work/DSA/c++/test1/maximumProfitOnApp.cpp
#include <bits/stdc++.h>
using namespace std;

int maximumProfit(int input[], int n)
{
    sort(input, input + n);
    int max = 0;
    for (int i = 0; i < n; i++)
    {
        if (max < input[i] * (n - i))
        {
            max = input[i] * (n - i);
        }
    }

    return max;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n, *input, i, *cost;
    cin >> n;
    input = new int[n];
    for (i = 0; i < n; i++)
        cin >> input[i];

    cout << maximumProfit(input, n) << endl;
}

File: work/DSA/javascript/test1/maximumProfitOnApp.js
function maximumProfit(input, size) {
    input.sort((a, b) => a - b)
    let max = 0
    for (let i = 0; i < size; i++) {
        if (max < input[i] * (size - i)) {
            max = input[i] * (size - i)
        }
    }
    return max;
}


const readline = require("readline-sync")
let size = Number(readline.question());
let input = readline.question().split(" ").map(x => parseInt(x))
let answer = maximumProfit(input, size)
console.log(answer)